lib/commands/ci.js

```diff
@@ -75,14 +75,16 @@ class CI extends ArboristWorkspaceCmd {
       )
     }
 
-    // Only remove node_modules after we've successfully loaded the virtual
-    // tree and validated the lockfile
-    await this.npm.time('npm-ci:rm', async () => {
-      const path = `${where}/node_modules`
-      // get the list of entries so we can skip the glob for performance
-      const entries = await fs.readdir(path, null).catch(er => [])
-      return Promise.all(entries.map(f => fs.rm(`${path}/${f}`, { force: true, recursive: true })))
-    })
+    if (!this.npm.flatOptions.dryRun) {
+      // Only remove node_modules after we've successfully loaded the virtual
+      // tree and validated the lockfile
+      await this.npm.time('npm-ci:rm', async () => {
+        const path = `${where}/node_modules`
+        // get the list of entries so we can skip the glob for performance
+        const entries = await fs.readdir(path, null).catch(er => [])
+        return Promise.all(entries.map(f => fs.rm(`${path}/${f}`, { force: true, recursive: true })))
+      })
+    }
```

Comment:
maybe it'd be better if, when a dry run, it checked that npm has *permissions* to rimraf node_modules?

Reply:
I am not sure what you mean by 'permissions'. Sorry this is my first time contributing to npm.

Reply:
You’re right, but i meant it’d be useful and match the “dry run” semantics

Reply:
Checking permissions in node is mostly done by trying the operation and then reporting to the user if it failed.  It's even discouraged for things like `fs.access` to be used for this purpose.  With that in mind I think it's fine for `dry-run` to simply tell the user what it would try to do.  It is not meant to be a test of if npm will eventually be able to succeed.  Many other things could go wrong (i.e. running out of disk space).

This PR appears fine as-is (I am not done fully reviewing it but at a glance it appears mostly on the right track).

Reply:
presumably *deleting* files wouldn't cause disk space to run out :-)

def that enhancement could be done as a follow through regardless